yb-voyager/cmd/exportSchema.go

```diff
@@ -226,27 +225,37 @@ func applyMigrationAssessmentRecommendations() error {
 	}
 
 	log.Infof("parsing assessment report json file for applying recommendations")
-	var report AssessmentReport
-	err := jsonfile.NewJsonFile[AssessmentReport](assessmentReportPath).Load(&report)
+	report, err := ParseJSONToAssessmentReport(assessmentReportPath)
 	if err != nil {
 		return fmt.Errorf("failed to parse json report file %q: %w", assessmentReportPath, err)
 	}
 
-	err = applyColocatedVsShardedTableRecommendation(report.Sizing)
+	shardedTables, err := report.GetShardedTablesRecommendation()
 	if err != nil {
-		return fmt.Errorf("failed to apply colocated vs sharded table recommendation: %w", err)
+		return fmt.Errorf("failed to fetch sharded tables recommendation: %w", err)
+	} else {
+		err := applyShardedTablesRecommendation(shardedTables)
+		if err != nil {
+			return fmt.Errorf("failed to apply colocated vs sharded table recommendation: %w", err)
+		}
 	}
+
 	return nil
 }
 
-func applyColocatedVsShardedTableRecommendation(shardingReport *migassessment.SizingAssessmentReport) error {
+func applyShardedTablesRecommendation(shardedTables []string) error {
+	if shardedTables == nil {
+		log.Info("list of sharded tables is null hence all the tables are recommended as colocated")
+		return nil
+	}
+
 	filePath := utils.GetObjectFilePath(schemaDir, "TABLE")
 	if !utils.FileOrFolderExists(filePath) {
-		log.Warnf("required schema file %s does not exists, returning without applying the recommendations", filePath)
+		utils.PrintAndLog("Required schema file %s does not exists, returning without applying Colocated/Sharded Tables recommendation", filePath)
```

Comment:
Under what circumstances would the table file not exist?

Reply:
in case there is no table on the source, just some non-table objects like UDF or something, which is rare and might not be the real world.